WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PageListEventListener.kt

```diff
@@ -0,0 +1,226 @@
+                                "Error updating the post with type: ${event.error.type} and" +
+                                        " message: ${event.error.message}"
+                        )
+                    } else {
+                        handlePostUploadedWithoutError.invoke(
+                                RemoteId((event.causeOfChange as UpdatePost).remotePostId)
+                        )
+                        invalidateUploadStatus.invoke(
+                                listOf(LocalId((event.causeOfChange as CauseOfOnPostChanged.UpdatePost).localPostId))
+                        )
+                    }
+                }
+            }
+        }
+    }
+
+    @Suppress("unused")
+    @Subscribe(threadMode = BACKGROUND)
+    fun onMediaChanged(event: OnMediaChanged) {
+        if (!event.isError && event.mediaList != null) {
+            uploadStatusChanged(*event.mediaList.map { LocalId(it.localPostId) }.toTypedArray())
+        }
+    }
+
+    @Suppress("unused")
+    @Subscribe(threadMode = BACKGROUND)
+    fun onPostUploaded(event: OnPostUploaded) {
+        if (event.post != null && event.post.localSiteId == site.id) {
+            uploadStatusChanged(LocalId(event.post.id))
+            if (!event.isError) {
+                handlePostUploadedWithoutError.invoke(RemoteId(event.post.remotePostId))
+            }
+        }
+    }
+
+    @Suppress("unused")
+    @Subscribe(threadMode = BACKGROUND)
+    fun onMediaUploaded(event: OnMediaUploaded) {
+        if (event.isError || event.canceled) {
+            return
+        }
+        if (event.media == null || event.media.localPostId == 0 || site.id != event.media.localSiteId) {
+            // Not interested in media not attached to posts or not belonging to the current site
+            return
+        }
+        uploadStatusChanged(LocalId(event.media.localPostId))
+    }
+
+    /**
+     * Upload started, reload so correct status on uploading post appears
+     */
+    @Suppress("unused")
+    @Subscribe(threadMode = BACKGROUND)
+    fun onEventBackgroundThread(event: PostEvents.PostUploadStarted) {
+        if (!event.post.isPage) {
+            return
+        }
+
+        if (site.id == event.post.localSiteId) {
+            uploadStatusChanged(LocalId(event.post.id))
+
+            handlePostUploadedStarted(RemoteId(event.post.remotePostId))
+        }
+    }
+
+    /**
+     * Upload cancelled (probably due to failed media), reload so correct status on uploading post appears
+     */
+    @Suppress("unused")
+    @Subscribe(threadMode = BACKGROUND)
+    fun onEventBackgroundThread(event: PostEvents.PostUploadCanceled) {
+        if (site.id == event.post.localSiteId) {
+            uploadStatusChanged(LocalId(event.post.id))
+        }
+    }
+
+    @Suppress("unused")
+    @Subscribe(threadMode = BACKGROUND)
+    fun onEventBackgroundThread(event: VideoOptimizer.ProgressEvent) {
+        uploadStatusChanged(LocalId(event.media.localPostId))
```

Comment:
Thanks! Upon looking at the events again I am thinking I should also check in each event if it's a page as well since posts could be enqueued and the events could be propagated to these methods. 

`if (site.id == event.post.localSiteId && event.post.isPage) {`

Reply:
Resolved in 7fb606e

Reply:
Thanks! What about `fun onEventBackgroundThread(event: UploadService.UploadMediaRetryEvent)`. Shouldn't we also filter out mediaModels which belong to a different site? Wdyt?

Reply:
>if it's a page as well since posts could be enqueued and the events could be propagated to these methods

Good idea! We might want to create a separate PR and add `!isPage` check to `PostListEventListener`, wdyt?

Reply:
Yes, that's a really good idea!

Reply:
Issue created here https://github.com/wordpress-mobile/WordPress-Android/issues/11259

Reply:
👍 Thanks!